{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "16e721d0",
   "metadata": {},
   "source": [
    "I downloaded the bioinformatics files from the [stackexchange archives](https://archive.org/download/stackexchange) and pulled out the 'posts.xml' file, renaming it to 'bioinformatics_Posts.xml."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "084ff1d4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['bioinformatics_Posts.xml']"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "xml_files = [f for f in os.listdir() if f.endswith('xml')]\n",
    "xml_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "cc7067e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import xml.etree.ElementTree as ET\n",
    "import html # for html.unescape()\n",
    "import re\n",
    "from bs4 import BeautifulSoup # for soup.get_text()\n",
    "import pandas as pd\n",
    "\n",
    "def safe_unescape(html_raw):\n",
    "    if str(type(html_raw)) == \"<class 'NoneType'>\":\n",
    "        html_raw = ''\n",
    "    return html.unescape(html_raw)\n",
    "\n",
    "\n",
    "def list_tags(tags_str):\n",
    "    return ';'.join(tags_str[1:-1].split('><'))\n",
    "\n",
    "\n",
    "def get_html_text(html_raw):\n",
    "    html_str = safe_unescape(html_raw)\n",
    "    soup = BeautifulSoup(html_str, 'html.parser')\n",
    "    html_txt = soup.get_text()\n",
    "    return html_txt\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "703365dc",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def extract_posts(xml_file):\n",
    "    topic = xml_file.split('_')[0]\n",
    "    row_count = 0\n",
    "\n",
    "    with open(xml_file, encoding='utf-8') as xml_file_handle:\n",
    "        csv_file = xml_file.replace('.xml', '.csv')\n",
    "        # print(csv_file)\n",
    "        data_rows = []\n",
    "        for line in xml_file_handle:\n",
    "                if re.match('  <row ', line):\n",
    "                    row_count += 1\n",
    "                    row_root = ET.fromstring(line)\n",
    "\n",
    "                    title = get_html_text( row_root.attrib.get('Title') )\n",
    "                    body = get_html_text( row_root.attrib.get('Body') )\n",
    "                    tags = list_tags(safe_unescape( row_root.attrib.get('Tags') ) )\n",
    "                    last_activity_date = row_root.attrib.get('LastActivityDate')\n",
    "                    row_dict = {\n",
    "                        'topic':topic, \n",
    "                        'title':title, \n",
    "                        'body':body,\n",
    "                        'text': title + ': ' + body,\n",
    "                        'tags':tags, \n",
    "                        'last_activity_date':last_activity_date,\n",
    "                    }\n",
    "                    if (len(row_dict['tags']) > 0):\n",
    "                        data_rows.append(row_dict)\n",
    "                \n",
    "        return pd.DataFrame(data_rows) #.to_csv(csv_file, index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "4466b761",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/4t/7982ncgx1271113b4hlqb9jr0000gn/T/ipykernel_92940/2722553304.py:19: MarkupResemblesLocatorWarning: The input looks more like a filename than markup. You may want to open this file and pass the filehandle into Beautiful Soup.\n",
      "  soup = BeautifulSoup(html_str, 'html.parser')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>topic</th>\n",
       "      <th>title</th>\n",
       "      <th>body</th>\n",
       "      <th>text</th>\n",
       "      <th>tags</th>\n",
       "      <th>last_activity_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>What's the most efficient file format for the ...</td>\n",
       "      <td>I'd like to learn which format is most commonl...</td>\n",
       "      <td>What's the most efficient file format for the ...</td>\n",
       "      <td>human-genome|storage|file-formats</td>\n",
       "      <td>2022-02-23T16:01:18.060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>Accuracy of the original human DNA datasets se...</td>\n",
       "      <td>The Human Genome Project was the project of 'd...</td>\n",
       "      <td>Accuracy of the original human DNA datasets se...</td>\n",
       "      <td>hgp|phylogenetics</td>\n",
       "      <td>2017-12-28T11:43:48.257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>Mapping drug names to ATC codes</td>\n",
       "      <td>I'm interested working with the medication inf...</td>\n",
       "      <td>Mapping drug names to ATC codes: I'm intereste...</td>\n",
       "      <td>drugs|ontology</td>\n",
       "      <td>2017-05-16T19:36:42.037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>What are the optimal parameters for docking a ...</td>\n",
       "      <td>I'm looking to dock a large ligand (~90kDa) to...</td>\n",
       "      <td>What are the optimal parameters for docking a ...</td>\n",
       "      <td>proteins|docking</td>\n",
       "      <td>2017-05-18T00:01:22.427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>What is the difference between FASTA, FASTQ, a...</td>\n",
       "      <td>I'd like to learn the differences between 3 co...</td>\n",
       "      <td>What is the difference between FASTA, FASTQ, a...</td>\n",
       "      <td>fasta|fastq|file-formats|sam</td>\n",
       "      <td>2022-08-18T18:58:12.410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6088</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>How to do post hoc comparisons after a repeate...</td>\n",
       "      <td>I have a data set of several samples with thei...</td>\n",
       "      <td>How to do post hoc comparisons after a repeate...</td>\n",
       "      <td>statistics|modelling|graphs|lme4</td>\n",
       "      <td>2024-04-01T17:10:57.933</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6089</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>How do I get the gene annotations as a text fi...</td>\n",
       "      <td>Also posted on biostars\\nI have a genbank file...</td>\n",
       "      <td>How do I get the gene annotations as a text fi...</td>\n",
       "      <td>gene|sequence-annotation</td>\n",
       "      <td>2024-03-29T17:59:05.480</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6090</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>I'm trying to run aTRAM tool for assembly, but...</td>\n",
       "      <td>Code:\\n$ python3 atram.py --max-processes=12 \\...</td>\n",
       "      <td>I'm trying to run aTRAM tool for assembly, but...</td>\n",
       "      <td>ngs|assembly</td>\n",
       "      <td>2024-04-04T20:50:57.923</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6091</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>Dante Labs VCF: UPDATE: SOLVED</td>\n",
       "      <td>Has anyone tried to analyse the VCF files prov...</td>\n",
       "      <td>Dante Labs VCF: UPDATE: SOLVED: Has anyone tri...</td>\n",
       "      <td>assembly|chromosomes</td>\n",
       "      <td>2024-03-30T15:55:38.210</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6092</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>Trying to plot the results of GSEA in python</td>\n",
       "      <td>I'm having some major chaos with the output ta...</td>\n",
       "      <td>Trying to plot the results of GSEA in python: ...</td>\n",
       "      <td>python|gsea|pandas|gene-ontology|go</td>\n",
       "      <td>2024-03-31T02:36:25.460</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6093 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               topic                                              title  \\\n",
       "0     bioinformatics  What's the most efficient file format for the ...   \n",
       "1     bioinformatics  Accuracy of the original human DNA datasets se...   \n",
       "2     bioinformatics                    Mapping drug names to ATC codes   \n",
       "3     bioinformatics  What are the optimal parameters for docking a ...   \n",
       "4     bioinformatics  What is the difference between FASTA, FASTQ, a...   \n",
       "...              ...                                                ...   \n",
       "6088  bioinformatics  How to do post hoc comparisons after a repeate...   \n",
       "6089  bioinformatics  How do I get the gene annotations as a text fi...   \n",
       "6090  bioinformatics  I'm trying to run aTRAM tool for assembly, but...   \n",
       "6091  bioinformatics                     Dante Labs VCF: UPDATE: SOLVED   \n",
       "6092  bioinformatics       Trying to plot the results of GSEA in python   \n",
       "\n",
       "                                                   body  \\\n",
       "0     I'd like to learn which format is most commonl...   \n",
       "1     The Human Genome Project was the project of 'd...   \n",
       "2     I'm interested working with the medication inf...   \n",
       "3     I'm looking to dock a large ligand (~90kDa) to...   \n",
       "4     I'd like to learn the differences between 3 co...   \n",
       "...                                                 ...   \n",
       "6088  I have a data set of several samples with thei...   \n",
       "6089  Also posted on biostars\\nI have a genbank file...   \n",
       "6090  Code:\\n$ python3 atram.py --max-processes=12 \\...   \n",
       "6091  Has anyone tried to analyse the VCF files prov...   \n",
       "6092  I'm having some major chaos with the output ta...   \n",
       "\n",
       "                                                   text  \\\n",
       "0     What's the most efficient file format for the ...   \n",
       "1     Accuracy of the original human DNA datasets se...   \n",
       "2     Mapping drug names to ATC codes: I'm intereste...   \n",
       "3     What are the optimal parameters for docking a ...   \n",
       "4     What is the difference between FASTA, FASTQ, a...   \n",
       "...                                                 ...   \n",
       "6088  How to do post hoc comparisons after a repeate...   \n",
       "6089  How do I get the gene annotations as a text fi...   \n",
       "6090  I'm trying to run aTRAM tool for assembly, but...   \n",
       "6091  Dante Labs VCF: UPDATE: SOLVED: Has anyone tri...   \n",
       "6092  Trying to plot the results of GSEA in python: ...   \n",
       "\n",
       "                                     tags       last_activity_date  \n",
       "0       human-genome|storage|file-formats  2022-02-23T16:01:18.060  \n",
       "1                       hgp|phylogenetics  2017-12-28T11:43:48.257  \n",
       "2                          drugs|ontology  2017-05-16T19:36:42.037  \n",
       "3                        proteins|docking  2017-05-18T00:01:22.427  \n",
       "4            fasta|fastq|file-formats|sam  2022-08-18T18:58:12.410  \n",
       "...                                   ...                      ...  \n",
       "6088     statistics|modelling|graphs|lme4  2024-04-01T17:10:57.933  \n",
       "6089             gene|sequence-annotation  2024-03-29T17:59:05.480  \n",
       "6090                         ngs|assembly  2024-04-04T20:50:57.923  \n",
       "6091                 assembly|chromosomes  2024-03-30T15:55:38.210  \n",
       "6092  python|gsea|pandas|gene-ontology|go  2024-03-31T02:36:25.460  \n",
       "\n",
       "[6093 rows x 6 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "post_data = extract_posts('bioinformatics_Posts.xml')\n",
    "post_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4735f7f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "post_data.to_csv('bioinformatics_posts.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e1652bf7",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/rmhorton/miniforge3/envs/nlp/lib/python3.12/site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.\n",
      "  warnings.warn(\n",
      "/Users/rmhorton/miniforge3/envs/nlp/lib/python3.12/site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "# %pip install sentence-transformers\n",
    "\n",
    "from sentence_transformers import SentenceTransformer\n",
    "sentxformer = SentenceTransformer('sentence-transformers/all-mpnet-base-v2')\n",
    "\n",
    "post_data['vector'] = sentxformer.encode(post_data['text'].values).tolist()\n",
    "    \n",
    "post_data.to_parquet('bioinformatics_posts_feaurized.parquet')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b01d25a4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>topic</th>\n",
       "      <th>title</th>\n",
       "      <th>body</th>\n",
       "      <th>text</th>\n",
       "      <th>tags</th>\n",
       "      <th>last_activity_date</th>\n",
       "      <th>vector</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>What's the most efficient file format for the ...</td>\n",
       "      <td>I'd like to learn which format is most commonl...</td>\n",
       "      <td>What's the most efficient file format for the ...</td>\n",
       "      <td>human-genome|storage|file-formats</td>\n",
       "      <td>2022-02-23T16:01:18.060</td>\n",
       "      <td>[0.01382493320852518, -0.001170836272649467, -...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>Accuracy of the original human DNA datasets se...</td>\n",
       "      <td>The Human Genome Project was the project of 'd...</td>\n",
       "      <td>Accuracy of the original human DNA datasets se...</td>\n",
       "      <td>hgp|phylogenetics</td>\n",
       "      <td>2017-12-28T11:43:48.257</td>\n",
       "      <td>[0.04499347507953644, 0.06613805145025253, -0....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>Mapping drug names to ATC codes</td>\n",
       "      <td>I'm interested working with the medication inf...</td>\n",
       "      <td>Mapping drug names to ATC codes I'm interested...</td>\n",
       "      <td>drugs|ontology</td>\n",
       "      <td>2017-05-16T19:36:42.037</td>\n",
       "      <td>[-0.012040533125400543, 0.025775786489248276, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>What are the optimal parameters for docking a ...</td>\n",
       "      <td>I'm looking to dock a large ligand (~90kDa) to...</td>\n",
       "      <td>What are the optimal parameters for docking a ...</td>\n",
       "      <td>proteins|docking</td>\n",
       "      <td>2017-05-18T00:01:22.427</td>\n",
       "      <td>[0.00028346164617687464, -0.06295959651470184,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>What is the difference between FASTA, FASTQ, a...</td>\n",
       "      <td>I'd like to learn the differences between 3 co...</td>\n",
       "      <td>What is the difference between FASTA, FASTQ, a...</td>\n",
       "      <td>fasta|fastq|file-formats|sam</td>\n",
       "      <td>2022-08-18T18:58:12.410</td>\n",
       "      <td>[0.009203458204865456, -0.10056201368570328, 0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6088</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>How to do post hoc comparisons after a repeate...</td>\n",
       "      <td>I have a data set of several samples with thei...</td>\n",
       "      <td>How to do post hoc comparisons after a repeate...</td>\n",
       "      <td>statistics|modelling|graphs|lme4</td>\n",
       "      <td>2024-04-01T17:10:57.933</td>\n",
       "      <td>[-0.022957181558012962, -0.0423324778676033, 4...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6089</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>How do I get the gene annotations as a text fi...</td>\n",
       "      <td>Also posted on biostars\\nI have a genbank file...</td>\n",
       "      <td>How do I get the gene annotations as a text fi...</td>\n",
       "      <td>gene|sequence-annotation</td>\n",
       "      <td>2024-03-29T17:59:05.480</td>\n",
       "      <td>[-0.021438077092170715, -0.06102913245558739, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6090</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>I'm trying to run aTRAM tool for assembly, but...</td>\n",
       "      <td>Code:\\n$ python3 atram.py --max-processes=12 \\...</td>\n",
       "      <td>I'm trying to run aTRAM tool for assembly, but...</td>\n",
       "      <td>ngs|assembly</td>\n",
       "      <td>2024-04-04T20:50:57.923</td>\n",
       "      <td>[0.0094567546620965, -0.04444451630115509, -0....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6091</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>Dante Labs VCF: UPDATE: SOLVED</td>\n",
       "      <td>Has anyone tried to analyse the VCF files prov...</td>\n",
       "      <td>Dante Labs VCF: UPDATE: SOLVED Has anyone trie...</td>\n",
       "      <td>assembly|chromosomes</td>\n",
       "      <td>2024-03-30T15:55:38.210</td>\n",
       "      <td>[0.03195073455572128, -0.06578633189201355, 0....</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6092</th>\n",
       "      <td>bioinformatics</td>\n",
       "      <td>Trying to plot the results of GSEA in python</td>\n",
       "      <td>I'm having some major chaos with the output ta...</td>\n",
       "      <td>Trying to plot the results of GSEA in python I...</td>\n",
       "      <td>python|gsea|pandas|gene-ontology|go</td>\n",
       "      <td>2024-03-31T02:36:25.460</td>\n",
       "      <td>[0.021045345813035965, 0.04397245869040489, 0....</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6093 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               topic                                              title  \\\n",
       "0     bioinformatics  What's the most efficient file format for the ...   \n",
       "1     bioinformatics  Accuracy of the original human DNA datasets se...   \n",
       "2     bioinformatics                    Mapping drug names to ATC codes   \n",
       "3     bioinformatics  What are the optimal parameters for docking a ...   \n",
       "4     bioinformatics  What is the difference between FASTA, FASTQ, a...   \n",
       "...              ...                                                ...   \n",
       "6088  bioinformatics  How to do post hoc comparisons after a repeate...   \n",
       "6089  bioinformatics  How do I get the gene annotations as a text fi...   \n",
       "6090  bioinformatics  I'm trying to run aTRAM tool for assembly, but...   \n",
       "6091  bioinformatics                     Dante Labs VCF: UPDATE: SOLVED   \n",
       "6092  bioinformatics       Trying to plot the results of GSEA in python   \n",
       "\n",
       "                                                   body  \\\n",
       "0     I'd like to learn which format is most commonl...   \n",
       "1     The Human Genome Project was the project of 'd...   \n",
       "2     I'm interested working with the medication inf...   \n",
       "3     I'm looking to dock a large ligand (~90kDa) to...   \n",
       "4     I'd like to learn the differences between 3 co...   \n",
       "...                                                 ...   \n",
       "6088  I have a data set of several samples with thei...   \n",
       "6089  Also posted on biostars\\nI have a genbank file...   \n",
       "6090  Code:\\n$ python3 atram.py --max-processes=12 \\...   \n",
       "6091  Has anyone tried to analyse the VCF files prov...   \n",
       "6092  I'm having some major chaos with the output ta...   \n",
       "\n",
       "                                                   text  \\\n",
       "0     What's the most efficient file format for the ...   \n",
       "1     Accuracy of the original human DNA datasets se...   \n",
       "2     Mapping drug names to ATC codes I'm interested...   \n",
       "3     What are the optimal parameters for docking a ...   \n",
       "4     What is the difference between FASTA, FASTQ, a...   \n",
       "...                                                 ...   \n",
       "6088  How to do post hoc comparisons after a repeate...   \n",
       "6089  How do I get the gene annotations as a text fi...   \n",
       "6090  I'm trying to run aTRAM tool for assembly, but...   \n",
       "6091  Dante Labs VCF: UPDATE: SOLVED Has anyone trie...   \n",
       "6092  Trying to plot the results of GSEA in python I...   \n",
       "\n",
       "                                     tags       last_activity_date  \\\n",
       "0       human-genome|storage|file-formats  2022-02-23T16:01:18.060   \n",
       "1                       hgp|phylogenetics  2017-12-28T11:43:48.257   \n",
       "2                          drugs|ontology  2017-05-16T19:36:42.037   \n",
       "3                        proteins|docking  2017-05-18T00:01:22.427   \n",
       "4            fasta|fastq|file-formats|sam  2022-08-18T18:58:12.410   \n",
       "...                                   ...                      ...   \n",
       "6088     statistics|modelling|graphs|lme4  2024-04-01T17:10:57.933   \n",
       "6089             gene|sequence-annotation  2024-03-29T17:59:05.480   \n",
       "6090                         ngs|assembly  2024-04-04T20:50:57.923   \n",
       "6091                 assembly|chromosomes  2024-03-30T15:55:38.210   \n",
       "6092  python|gsea|pandas|gene-ontology|go  2024-03-31T02:36:25.460   \n",
       "\n",
       "                                                 vector  \n",
       "0     [0.01382493320852518, -0.001170836272649467, -...  \n",
       "1     [0.04499347507953644, 0.06613805145025253, -0....  \n",
       "2     [-0.012040533125400543, 0.025775786489248276, ...  \n",
       "3     [0.00028346164617687464, -0.06295959651470184,...  \n",
       "4     [0.009203458204865456, -0.10056201368570328, 0...  \n",
       "...                                                 ...  \n",
       "6088  [-0.022957181558012962, -0.0423324778676033, 4...  \n",
       "6089  [-0.021438077092170715, -0.06102913245558739, ...  \n",
       "6090  [0.0094567546620965, -0.04444451630115509, -0....  \n",
       "6091  [0.03195073455572128, -0.06578633189201355, 0....  \n",
       "6092  [0.021045345813035965, 0.04397245869040489, 0....  \n",
       "\n",
       "[6093 rows x 7 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "post_data"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "nlp",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
